File: Client/src/type/product/ProductResponse.ts
import type { PageResponse } from "../../api/BaseApi/baseApi";

export interface ProductResponse {
  id: string;
  name: string;
  description: string;
  price: number;
  path: string;
  img: string;
  hoverImg?: string;
  productTypeId: number;
}

export type ProductResponsePageResponse = PageResponse<ProductResponse>;

File: src/main/java/com/example/Server/dto/response/product/ProductResponse.java
package com.example.Server.dto.response.product;

import lombok.*;

@Data @NoArgsConstructor @AllArgsConstructor
public class ProductResponse {
    private String id;
    private String name;
    private String description;
    private Double price;
    private String path;
    private String img;
    private String hoverImg;
    private Long productTypeId;
}
